src/System.Linq/src/System/Linq/Union.cs

```diff
@@ -165,6 +165,11 @@ public List<TSource> ToList()
                 return FillSet().ToList();
             }
 
+            public HashSet<TSource> ToHashSet(IEqualityComparer<TSource> comparer)
+            {
+                return FillSet().ToHashSet(comparer);
```

Comment:
You can replace this with `new HashSet<TSource>(this, comparer)` and delete the Set method